### PR TITLE
dbeaver/pro#2941 Fix new table creation

### PR DESCRIPTION
**Pull request by E1izabeth**
Should not be reproducible.
Please test several different databases.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java [hunk 1]
```diff
@@ -222,6 +222,9 @@ protected IStatus run(DBRProgressMonitor monitor) {
                 return Status.CANCEL_STATUS;//GeneralUtils.makeErrorStatus("Null object returned");
             }
             monitor.beginTask("Save " + newObject.getClass().getSimpleName(), 3);
+            var closure = new Object() {
+                DBNDatabaseNode node = null;
+            };
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: Why not the common pattern with a single-element array?
```java
DBNDatabaseNode[] node = {null};
```

Your approach will result in the creation of a new inner anonymous class file.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: I don't like this pattern, it looks bad. Of course, if you insist, I can change it,

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: Also that's what IDE suggest :)

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: Any approach introducing a heap-allocated object with a reference will work. The array version is very common and doesn't introduce overhead like an anonymous class.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: OK

Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/actions/NavigatorHandlerObjectCreateBase.java]: I'd move it in the class variable, why use closure or array?

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Pull request merged**